### ISSUE TITLE
Matrix latex printing in the IPython notebook prints an Error

### ISSUE THREAD

**Issue by moorepants** [sympy/sympy#9629]
This error has been reported before and I thought that it was fixed, but we've had several people in the SciPy tutorial get caught by this. To reproduce you must not have LaTeX installed and call `init_printing` (in our case we were using `init_vprinting` from `sympy.physics.vector`) with no arguments for the defaults. This issue is that the printers print both a matplotlib based png and mathjax code behind the scenes with the default `init_printing` arguments. The matplotlib png creation will fail due to it not supporting some latex symbols for our matrix latex printer. And error is displayed for the printing, even though mathjax will properly print the correct result. Here is a screenshot of the notebook:
![selection_105](https://cloud.githubusercontent.com/assets/276007/8555384/68866c98-24b5-11e5-97d3-ba6b35d07941.png)

and the full traceback:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/home/ton/miniconda/lib/python2.7/site-packages/IPython/core/formatters.pyc in __call__(self, obj)
    328                 pass
    329             else:
--> 330                 return printer(obj)
    331             # Finally look for special method names
    332             method = _safe_get_formatter_method(obj, self.print_method)

/home/ton/miniconda/lib/python2.7/site-packages/sympy/interactive/printing.pyc in _print_latex_png(o)
    123                 if latex_mode != 'inline':
    124                     s = latex(o, mode='inline')
--> 125                 return _matplotlib_wrapper(s)
    126 
    127     def _print_latex_matplotlib(o):

/home/ton/miniconda/lib/python2.7/site-packages/sympy/interactive/printing.pyc in _matplotlib_wrapper(o)
     86         o = o.replace(r'\operatorname', '')
     87         o = o.replace(r'\overline', r'\bar')
---> 88         return latex_to_png(o)
     89 
     90     def _can_print_latex(o):

/home/ton/miniconda/lib/python2.7/site-packages/IPython/lib/latextools.pyc in latex_to_png(s, encode, backend, wrap)
     85     else:
     86         raise ValueError('No such backend {0}'.format(backend))
---> 87     bin_data = f(s, wrap)
     88     if encode and bin_data:
     89         bin_data = encodestring(bin_data)

/home/ton/miniconda/lib/python2.7/site-packages/IPython/lib/latextools.pyc in latex_to_png_mpl(s, wrap)
    104     mt = mathtext.MathTextParser('bitmap')
    105     f = BytesIO()
--> 106     mt.to_png(f, s, fontsize=12)
    107     return f.getvalue()
    108 

/home/ton/miniconda/lib/python2.7/site-packages/matplotlib/mathtext.pyc in to_png(self, filename, texstr, color, dpi, fontsize)
   3092         image in pixels.
   3093         """
-> 3094         rgba, depth = self.to_rgba(texstr, color=color, dpi=dpi, fontsize=fontsize)
   3095         numrows, numcols, tmp = rgba.shape
   3096         _png.write_png(rgba.tostring(), numcols, numrows, filename)

/home/ton/miniconda/lib/python2.7/site-packages/matplotlib/mathtext.pyc in to_rgba(self, texstr, color, dpi, fontsize)
   3057             image in pixels.
   3058         """
-> 3059         x, depth = self.to_mask(texstr, dpi=dpi, fontsize=fontsize)
   3060 
   3061         r, g, b = mcolors.colorConverter.to_rgb(color)

/home/ton/miniconda/lib/python2.7/site-packages/matplotlib/mathtext.pyc in to_mask(self, texstr, dpi, fontsize)
   3030         assert(self._output=="bitmap")
   3031         prop = FontProperties(size=fontsize)
-> 3032         ftimage, depth = self.parse(texstr, dpi=dpi, prop=prop)
   3033 
   3034         x = ftimage.as_array()

/home/ton/miniconda/lib/python2.7/site-packages/matplotlib/mathtext.pyc in parse(self, s, dpi, prop)
   3003             self.__class__._parser = Parser()
   3004 
-> 3005         box = self._parser.parse(s, font_output, fontsize, dpi)
   3006         font_output.set_canvas_size(box.width, box.height, box.depth)
   3007         result = font_output.get_results(box)

/home/ton/miniconda/lib/python2.7/site-packages/matplotlib/mathtext.pyc in parse(self, s, fonts_object, fontsize, dpi)
   2337                         err.line,
   2338                         " " * (err.column - 1) + "^",
-> 2339                         six.text_type(err)]))
   2340         self._state_stack = None
   2341         self._em_width_cache = {}

ValueError: 
\left[\begin{smallmatrix}1 & 0 & 0 & 0 & 0 & 0\\0 & 1 & 0 & 0 & 0 & 0\\0 & 0 & 1 & 0 & 0 & 0\\0 & 0 & 0 & m_{0} + m_{1} + m_{2} & - l_{0} \left(m_{1} + m_{2}\right) {sin}\left(q_{1}\right) & - l_{1} m_{2} {sin}\left(q_{2}\right)\\0 & 0 & 0 & - l_{0} \left(m_{1} + m_{2}\right) {sin}\left(q_{1}\right) & l_{0}^{2} m_{1} + l_{0}^{2} m_{2} & l_{0} l_{1} m_{2} {cos}\left(q_{1} - q_{2}\right)\\0 & 0 & 0 & - l_{1} m_{2} {sin}\left(q_{2}\right) & l_{0} l_{1} m_{2} {cos}\left(q_{1} - q_{2}\right) & l_{1}^{2} m_{2}\end{smallmatrix}\right]
      ^
Expected "\right" (at char 6), (line:1, col:7)
```

**Comment by moorepants** [comment]
Ah, this is another report: https://github.com/sympy/sympy/issues/3006

**Comment by florianGithub** [comment]
same problem here.
Is there any solution?

**Comment by jksuom** [comment]
``matplotlib.mathtext`` is unable to deal with latex environments. One solution is to install latex and run ``init_printing(use_latex='png')``.

**Comment by moorepants** [comment]
If you want mathjax printing without installing LaTeX just use `init_printing(use_latex='mathjax')`.  Otherwise you need to install LaTeX to avoid this error.

We need to deal with this. It is quite annoying.

The fact that the notebook printer will run both mathjax and matplotlib printing in the background when it only needs to run mathjax is one problem.

The second problem is that we should give a better error message if a user tries to use matplotlib printing with these symbols. Or we need to add support to matplotlib for these missing tex symbols.

**Comment by moorepants** [comment]
Fixed in #9959.
